### DISCUSSION TITLE
Saving only one slice of 4d velocity field

### DISCUSSION THREAD

**Discussion by keli559** [CliMA/Oceananigans.jl#3292] [Helpdesk]
In the demo, I saw using the following code to collect the output. How would I do it if I only want to save one slice at certain x and y ? 
My velocities fields are 4d, time, depth, lat and lon. I only want to save time and depth with fixed lat and lon. 
fields_to_output = merge(model.velocities, model.tracers, (; νₑ, dbdz, dcdz))

**Comment by glwagner** [comment]
Hi @keli559! This feature is described in our documentation:

https://clima.github.io/OceananigansDocumentation/stable/model_setup/output_writers/#JLD2-output-writer

To save at some specific `x, y` you need to identify the `i, j` indices that you'd like to extract, and pass this to the output writer. I'm describing data at a given `x, y` as a "column" in physical space:

```julia
# Selected indices are i, j = 3, 4
ii = 3
jj = 4
column_indices = (ii, jj, :) # indices of a column with all points in z, and one point in x, y

# syntax is the same for `NetCDFOutputWriter`
output_writer = JLD2OutputWriter(model, outputs; indices=column_indices, other_kw...)
```

To select indices corresponding to a particular location you can use something like

```julia
x = xnodes(grid, Center(), Center(), Center())
x_of_interest = 4.1 # for example
ii = searchsortedfirst(x, x_of_interest)
```

This might need additional work for the GPU (or just `CUDA.@allowscalar`).

PS I'm converting this to a discussion so we can continue there; I don't think this issue motivates any changes to the source code.
